File: output/deepseek_chat/avatar/algo_based_translation/python/java/failed_translation/atcoder_AGC021_B.java

import java.util.*;
import java.lang.Math;

public class atcoder_AGC021_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        List<Integer> XYList = new ArrayList<>();
        while (scanner.hasNextInt()) {
            XYList.add(scanner.nextInt());
        }
        
        List<int[]> XY = new ArrayList<>();
        for (int i = 0; i < XYList.size(); i += 2) {
            int[] point = {XYList.get(i), XYList.get(i + 1)};
            XY.add(point);
        }
        
        for (int i = 0; i < XY.size(); i++) {
            int x = XY.get(i)[0];
            int y = XY.get(i)[1];
            List<Double> D = new ArrayList<>();
            
            for (int j = 0; j < XY.size(); j++) {
                if (j != i) {
                    int X = XY.get(j)[0];
                    int Y = XY.get(j)[1];
                    double angle = Math.atan2(Y - y, X - x);
                    D.add(angle);
                }
            }
            
            Collections.sort(D);
            D.add(D.get(0) + 2 * Math.PI);
            
            double ans = 0;
            for (int k = 0; k < D.size() - 1; k++) {
                double a = D.get(k);
                double b = D.get(k + 1);
                if (b - a >= Math.PI) {
                    ans = (b - a) - Math.PI;
                }
            }
            
            System.out.println(ans / (2 * Math.PI));
        }
    }
}

File: output/deepseek_chat/avatar/algo_based_translation/java/python/failed_translation/atcoder_AGC021_B.py

import math
import sys

def main():
    data = sys.stdin.read().split()
    idx = 0
    N = int(data[idx]); idx += 1
    x = [0] * N
    y = [0] * N
    for i in range(N):
        x[i] = int(data[idx]); idx += 1
        y[i] = int(data[idx]); idx += 1
    
    for i in range(N):
        thetas = []
        for j in range(N):
            if i == j:
                continue
            angle = math.atan2(y[j] - y[i], x[j] - x[i])
            thetas.append(angle)
        thetas.sort()
        thetas.append(thetas[0] + 2 * math.pi)
        ans = 0.0
        for k in range(N - 1):
            gap = thetas[k + 1] - thetas[k] - math.pi
            if gap > ans:
                ans = gap
        print(ans / (2 * math.pi))

if __name__ == "__main__":
    main()

